overwatch/src/overwatch/life_cycle.rs

```diff
@@ -85,21 +94,20 @@ impl ServicesLifeCycleHandle {
         }
         Ok(())
     }
-
-    /// Get all [`ServiceId`]s registered in this handle
-    pub fn services_ids(&self) -> impl Iterator<Item = &ServiceId> {
-        self.handlers.keys()
-    }
 }
 
-impl<const N: usize> TryFrom<[(ServiceId, LifecycleHandle); N]> for ServicesLifeCycleHandle {
+impl<const N: usize, AggregatedServiceId> TryFrom<[(AggregatedServiceId, LifecycleHandle); N]>
+    for ServicesLifeCycleHandle<AggregatedServiceId>
+where
+    AggregatedServiceId: Eq + Hash + Display,
+{
     // TODO: On errors refactor extract into a concrete error type with `thiserror`
     type Error = Box<dyn Error + Send + Sync>;
 
-    fn try_from(value: [(ServiceId, LifecycleHandle); N]) -> Result<Self, Self::Error> {
+    fn try_from(value: [(AggregatedServiceId, LifecycleHandle); N]) -> Result<Self, Self::Error> {
         let mut handlers = HashMap::new();
         for (service_id, handle) in value {
-            if handlers.contains_key(service_id) {
+            if handlers.contains_key(&service_id) {
```

Comment:
Nice to keep this just in case there is someone crazy enough to implement stuff from scratch.